inc/search.class.php

```diff
@@ -6590,6 +6590,38 @@ static function makeTextCriteria ($field, $val, $not = false, $link = 'AND') {
       return " $link ($sql $sql_or)";
    }
 
+   /**
+    * Create SQL search value
+    *
+    * @since 9.4
+    *
+    * @param string  $val value to search
+    *
+    * @return string|null
+   **/
+   static function makeTextSearchValue($val) {
+      // Unclean to permit < and > search
+      $val = Toolbox::unclean_cross_side_scripting_deep($val);
+
+      // escape _ char used as wildcard in mysql likes
+      $val = str_replace('_', '\\_', $val);
+
+      if (($val == 'NULL') || ($val == 'null')) {
+         return null;
+      }
+
+      $search = '';
+      preg_match('/^(\^)?([^\^\$]*)(\$)?$/', $val, $matches);
```

Comment:
This change would made matches 1, 2 and 3 always set.
```suggestion
      preg_match('/^(\^?)([^\^\$]*)(\$?)$/', $val, $matches);
```